src/program.ts

```diff
@@ -0,0 +1,286 @@
+   * @param space
+   * @returns
+   */
+  public async createStorageAccount(space: number): Promise<Keypair> {
+    const lamports = await this.connection.getMinimumBalanceForRentExemption(
+      space
+    );
+
+    let account = Keypair.generate();
+
+    const transaction = new Transaction().add(
+      SystemProgram.createAccount({
+        fromPubkey: this.payerAccount.publicKey,
+        newAccountPubkey: account.publicKey,
+        lamports,
+        space,
+        programId: this.programAccount.publicKey,
+      })
+    );
+
+    await sendAndConfirmTransaction(
+      this.connection,
+      transaction,
+      [this.payerAccount, account],
+      {
+        skipPreflight: false,
+        commitment: 'confirmed',
+        preflightCommitment: undefined,
+      }
+    );
+
+    return account;
+  }
+
+  /**
+   * Invokes the given callback on every program log.
+   *
+   * @param callback  The function to invoke whenever a `Program log:` is parsed in the logs.
+   */
+  public addLogListener(callback: LogCallback): number {
+    return this.logs.addLogListener(callback);
+  }
+
+  /**
+   * Unsubscribes from the given log listener id.
+   *
+   * @param listenerId The log listener id
+   */
+  public async removeLogListener(listenerId: number): Promise<void> {
+    return await this.logs.removeLogListener(listenerId);
+  }
+
+  /**
+   * Invokes the given callback every time the given event is emitted.
+   *
+   * @param abi       ABI interface
+   * @param callback  The function to invoke whenever the event is emitted from
+   *                  program logs.
+   */
+  public addEventListener(
+    abi: ethers.utils.Interface,
+    callback: EventCallback
+  ): number {
+    return this.logs.addEventListener(abi, callback);
+  }
+
+  /**
+   * Unsubscribes from the given eventName.
+   *
+   * @param listenerId The log listener id
+   */
+  public async removeEventListener(listenerId: number): Promise<void> {
+    return await this.logs.removeEventListener(listenerId);
+  }
+
+  /**
+   * Deploy a new contract in a loaded Solang program
+   *
+   * @param program
+   * @param options
+   * @returns
+   */
+  public async deployContract(
+    options: ContractDeployOptions
+  ): Promise<ContractDeployResult> {
+    return Contract.deploy(this, options);
+  }
+
+  /**
+   * Load a deployed contract
+   *
+   * @param program
+   * @param abiData
+   * @param contractStorageAccount
+   * @returns                      The contract instance
+   */
+  public async getContract(
+    abiData: string,
+    contractStorageAccount: Keypair
+  ): Promise<Contract> {
+    return Contract.get(this, abiData, contractStorageAccount);
+  }
+
+  /**
+   * Make and execute a transaction with the given `instruction` and `signers`
+   *
+   * @param simulate     Whether to perform a dry-run
+   * @param instruction  Solana instruction
+   * @param signers      List of signers
+   * @returns            Transaction result
+   */
+  public async makeTx(
```

Comment:
`sendTransaction` would be more consistent with existing APIs. Can you rely on there only being one instruction? Why not accept a `Transaction` instead?

Overall this method seems a bit long and could probably be broken up into two methods, one for simulation and one for sending, with different options passed to each.

Reply:
We need to be able to add more than one instruction to a transaction. For ed25519 signature verifications, an ed25519 instruction needs to be included (order doesn't matter).
Besides, multiple instructions per instructions is a great feature that should be exposed :)